mat64/dense.go

```diff
@@ -113,6 +113,31 @@ func (m *Dense) reuseAs(r, c int) {
 	}
 }
 
+// untranspose untransposes a matrix if applicable. If a is an Untransposer, then
+// untranspose returns the underlying matrix and true. If it is not, then it returns
+// the input matrix and false.
+func untranspose(a Matrix) (Matrix, bool) {
+	if ut, ok := a.(Untransposer); ok {
+		return ut.Untranspose(), true
+	}
+	return a, false
+}
+
+// uniqueMemoryDense returns m with different memory from a. It also returns
+// a callback to defer for cleanup at the return of the call if necessary.
+func uniqueMemoryDense(m *Dense, a Matrix) (*Dense, func()) {
```

Comment:
I agree. "With" was intended to indicate that you are working with the
second matrix, I was leaning towards "Considering" but didn't like
that. 

isolatedWorkspace and you have a deal - isolateWorkspace suggests
working in place (which we can't do).

Also add named returns, w and restore purely for documentation.